tests/test__collection_api.py

```diff
@@ -2170,6 +2170,14 @@ def test_find_with_comment(self):
         actual = list(self.db.collection.find({'_id': 1, '$comment': 'test'}))
         self.assertEqual([{'_id': 1}], actual)
 
+    def test__find_or_and(self):
+        self.db.collection.insert_many([
+            {'x': 1, 'y': 1},
+        ])
+        self.assertEqual(
+            [],
+            [d['x'] for d in self.db.collection.find({'y': 2, '$or': [{'x': 1}, {'x': 2}]})])
```

Comment:
Your test only fails randomly as the key of a dict are not sorted. I've replaced your search filter by an OrderedDict that forces the `$or` key to come first.